session/session.go

```diff
@@ -224,3 +241,26 @@ func getDefaultTransport(endpointURL string) TransportHandler {
 
 	return transportHandler
 }
+
+func isTimeout(err error) bool {
+	if slErr, ok := err.(sl.Error); ok {
+		switch slErr.StatusCode {
+		case 408, 504, 599:
+			return true
+		}
```

Comment:
continue here with checking the wrapped error (recursive call to isTimeout, or set err to slErr.Wrapped)